src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```
@@ -30,13 +30,29 @@ class ResourceService(private val resourceTypes: Map[ResourceTypeName, ResourceT
     accessPolicyDAO.createResourceType(resourceType.name)
   }
 
-  def createResource(resourceType: ResourceType, resourceId: ResourceId, userInfo: UserInfo): Future[Resource] = {
+  def createResource(resourceType: ResourceType, resourceId: ResourceId, policiesOpt: Option[Map[AccessPolicyName, AccessPolicyMembership]] = None, userInfo: UserInfo): Future[Resource] = {
     accessPolicyDAO.createResource(Resource(resourceType.name, resourceId)) flatMap { resource =>
-      val role = resourceType.roles.find(_.roleName == resourceType.ownerRoleName).getOrElse(throw new WorkbenchException(s"owner role ${resourceType.ownerRoleName} does not exist in $resourceType"))
-      val subjects: Set[WorkbenchSubject] = Set(userInfo.userId)
-      val resourceAndPolicyName = ResourceAndPolicyName(resource, AccessPolicyName(role.roleName.value))
-
-      createPolicy(resourceAndPolicyName, subjects, Set(role.roleName), Set.empty).map {_ => Resource(resourceType.name, resourceId)}
+      policiesOpt match {
+        case Some(policies) =>
+          Future.traverse(policies) { case (accessPolicyName, accessPolicyMembership) =>
+            val resourceAndPolicyName = ResourceAndPolicyName(Resource(resourceType.name, resourceId), accessPolicyName)
+            val loadedSubjects = Future.traverse(accessPolicyMembership.memberEmails) { directoryDAO.loadSubjectFromEmail }.map(subjOpts => subjOpts.collect { case Some(subj) => subj })
+            val roles = accessPolicyMembership.roles
+            val actions = accessPolicyMembership.actions
+
+            loadedSubjects.flatMap { subjects =>
+              createPolicy(resourceAndPolicyName, subjects, roles, actions)
```

https://broadinstitute.atlassian.net/browse/GAWB-2994

We used to not have to worry about validating policies on Resource create because the policies being created we precisely structured and populated.  Now, because we allow users to submit an arbitrary list of policies with whatever data they want in them, I think we need to validate the policies here before creating them.  

At the very least, it seems like line 39 is reintroducing bug GAWB-2994, in that it seems to just silently ignore email addresses that don't already map to a subject in SAM.